src/components/Button/Button.types.ts

```diff
@@ -0,0 +1,33 @@
+import { ReactElement } from "react";
+
+export enum COLOR_VARIANTS {
+  GREEN = 'green',
+  WHITE = 'white',
+}
+export enum FONT_VARIANTS {
+  STANDARD = 'standard',
+  UPPERCASE = 'uppercase',
+}
+export enum PURPOSE {
+  ANCHOR = 'anchor',
+  SUBMIT = 'submit',
+  FUNCTION_TRIGGER = 'functionTrigger'
+}
+
+export interface ButtonProps {
+  label: string,
+  colorVariant: COLOR_VARIANTS,
+  fontVariant: FONT_VARIANTS,
+  purpose: PURPOSE,
+  anchorLink?: string,
+  onclickAction?: () => void,
+}
+
+export type PurposeSelector = (
```

Comment:
`PurposeSelectorProps`

Reply:
That is a **function type, not argumets/props type**. I do not think I've done function types in a way You sugessted.